src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -568,3 +568,153 @@ def test_lfric_driver_field_array_inc():
+    external functions that use module variables.
+
+    '''
+    _, invoke = get_invoke("driver_creation/invoke_kernel_with_imported_"
+                           "symbols.f90", API, dist_mem=False, idx=0)
+
+    extract = LFRicExtractTrans()
+    extract.apply(invoke.schedule.children[0],
+                  options={"create_driver": True,
+                           "region_name": ("import", "test")})
+    code = str(invoke.gen())
+    assert ('CALL extract_psy_data%PreDeclareVariable("'
+            'module_var_a_post@module_with_var_mod", module_var_a)' in code)
+    assert ('CALL extract_psy_data%ProvideVariable("'
+            'module_var_a_post@module_with_var_mod", module_var_a)' in code)
+
+    filename = "driver-import-test.F90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    assert ("call extract_psy_data%ReadVariable('module_var_a_post@"
+            "module_with_var_mod', module_var_a_post)" in driver)
+    assert "if (module_var_a == module_var_a_post)" in driver
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir", "init_module_manager")
+def test_lfric_driver_external_symbols_name_clash():
+    '''Test the handling of symbols imported from other modules, or calls to
+    external functions that use module variables. In this example the external
+    module uses a variable with the same name as the user code, which causes
+    a name clash.
+
+    '''
+    _, invoke = get_invoke("driver_creation/invoke_kernel_with_imported_"
+                           "symbols.f90", API, dist_mem=False, idx=1)
+
+    extract = LFRicExtractTrans()
+    extract.apply(invoke.schedule.children[0],
+                  options={"create_driver": True,
+                           "region_name": ("import", "test")})
+    code = str(invoke.gen())
+
+    # Make sure the imported, clashing symbol 'f1_data' is renamed:
+    assert "USE module_with_name_clash_mod, ONLY: f1_data_1=>f1_data" in code
+    assert ('CALL extract_psy_data%PreDeclareVariable("f1_data@'
+            'module_with_name_clash_mod", f1_data_1)' in code)
+    assert ('CALL extract_psy_data%ProvideVariable("f1_data@'
+            'module_with_name_clash_mod", f1_data_1)' in code)
+
+    # Even though PSyclone cannot find the variable, it should still be
+    # extracted:
+
+    filename = "driver-import-test.F90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    assert ("call extract_psy_data%ReadVariable("
+            "'f1_data@module_with_name_clash_mod', f1_data_1)" in driver)
+    assert ("call extract_psy_data%ReadVariable("
+            "'f2_data@module_with_name_clash_mod', f2_data_1)" in driver)
+    assert ("call extract_psy_data%ReadVariable("
+            "'f2_data_post@module_with_name_clash_mod', f2_data_1_post)"
+            in driver)
```

Comment:
I've a feeling I've asked this before but is it possible to check that the generated driver compiles? If it is, please could you extend all relevant tests.

Reply:
Ooops, forgot. Yes, it is possible (except in one example where we test invalid Fortran code), and I've added it.

Reply:
Whoop!